### ISSUE TITLE
Incorrect decoding of `128u8`

### ISSUE THREAD

**Issue by noib3** [as-com/varint-simd#10]
```rust
#[test]
fn decode_128() {
    let (buf, len) = varint_simd::encode(128u8);
    assert_eq!(&buf[..len as usize], [128, 1]);

    // 💥 should ask for more bytes, instead returns Ok((0, 2)).
    assert!(matches!(
        varint_simd::decode::<u8>(&buf[..1]),
        Err(varint_simd::VarIntDecodeError::NotEnoughBytes)
    ));
}
```

**Comment by noib3** [comment]
Can you publish a new minor version?

**Comment by as-com** [comment]
Yep, I just published 0.4.1 with the fix
